views/backend/write.php

```diff
@@ -7,48 +7,54 @@
 <h2>Poster un article</h2>
 
 <?php
-    if(isset($_POST['post'])){
-        $title = htmlspecialchars(trim($_POST['title']));
-        $content = htmlspecialchars(trim($_POST['content']));
-        $posted = isset($_POST['public']) ? "1" : "0";
+   //Fonction permettant de vérifier les données envoyées par l'utilsiateur
+     function form_page_write(){
+        if(filter_has_var(INPUT_POST, 'post')){
+            if(filter_has_var(INPUT_POST, 'title')){
+                $title = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'title', FILTER_SANITIZE_STRING)));
+            }
+            if(isset($_POST['content'])){
+                $content = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'content', FILTER_SANITIZE_STRING)));
+            }
 
-        $errors = [];
+            $posted = filter_var(INPUT_POST, 'public') ? "1" : "0";
 
-        if(empty($title) || empty($content)){
-            $errors['empty'] = "Veuillez remplir tous les champs";
-        }
+            $errors = [];
 
-        if(!empty($_FILES['image']['name'])){
-            $file = $_FILES['image']['name'];
-            $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
-            $extension = strrchr($file,'.');
-            if(!in_array($extension,$extensions)){
-                $errors['image'] = "Cette image n'est pas valable";
+            if(empty($title) || empty($content)){
+                $errors['empty'] = "Veuillez remplir tous les champs";
             }
-        }
 
-        if(!empty($errors)){
-            ?>
-                <div class="card red">
-                    <div class="card-content white-text">
-                        <?php
-                            foreach($errors as $error){
-                                echo $error."<br/>";
-                            }
-                        ?>
-                    </div>
-                </div>
-            <?php
-        }else{
-            $model_article->post($title,$content,$posted);
             if(!empty($_FILES['image']['name'])){
-            
-                $model_article->post_img($_FILES['image']['tmp_name'], $extension);
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable";
+                }
+            }
 
+            if(!empty($errors)){
+                ?>
+                    <div class="card red">
+                        <div class="card-content white-text">
+                            <?php
+                                foreach($errors as $error){
+                                    echo $error;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$error'.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5344015)